server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -1585,6 +1586,12 @@ private enum ElasticsearchExceptionHandle {
             UnsupportedAggregationOnDownsampledIndex::new,
             167,
             Version.V_8_5_0
+        ),
+        SHARD_UNAVAILABLE_EXCEPTION(
+            ShardSearchFailure.ShardUnavailableException.class,
+            ShardSearchFailure.ShardUnavailableException::new,
+            168,
+            Version.V_8_7_0
```

Comment:
@javanna I know we talked about not having this, but apparently, every exception in /org/elasticsearch has to be registered here :/

See test: `org.elasticsearch.ExceptionSerializationTests.testExceptionRegistration`

Reply:
I see, I think the problem you are trying to work around is that the exception that gets thrown is used in different places, and you'd like to only affect the codepath where we wrap the different kinds of shard unavailable into the high-level one that gets returned to users. I think we should try to avoid adding yet another exception if possible. Would it be possible to extend the existing `NoShardAvailableActionException` to conditionally return an empty stacktrace, for instance based on an additional constructor argument? The flag would not need to be serialized, only used locally to control whether to expose empty stacktrace or the original one, while avoiding to create another exception instance.